dronekit_sitl/__init__.py

```diff
@@ -268,7 +347,23 @@ def cleanup_sim():
         # else:
         #     sitl = Popen(sitl_args, stdout=PIPE, stderr=PIPE)
 
-        p = Popen([self.path] + args, cwd=wd, shell=sys.platform == 'win32', stdout=PIPE, stderr=PIPE)
+        popen_args = []
+        if self.gdb:
```

Comment:
http://kfunk.org/2014/08/29/scripting-gdb-to-execute-commands-at-particular-breakpoints/

But we'd have to modify the interface to allow arbitrary scripting instead of just creating breakpoints.  Once gdb breaks you would basically have to kill the gdb process to continue, I think.